crates/llm-base/src/loader.rs

```diff
@@ -498,13 +553,13 @@ pub struct Loader<Hp: Hyperparameters, F: FnMut(LoadProgress)> {
 }
 impl<Hp: Hyperparameters, F: FnMut(LoadProgress)> Loader<Hp, F> {
     /// Creates a new loader.
-    pub fn new(load_progress_callback: F) -> Self {
+    pub fn new(vocabulary: Vocabulary, load_progress_callback: F) -> Self {
         Self {
             load_progress_callback,
 
             container_type: ContainerType::Ggml,
             hyperparameters: Hp::default(),
-            vocabulary: Vocabulary::default(),
+            vocabulary: vocabulary,
```

Comment:
Since the attribute name and variable name are the same, you can remove the attribute name, right? I believe that is the convention in the rest of the code.